### PR TITLE
[FRONT | API] Add Ability to Filter Objects by Creator (#issue1512)

### PR DESCRIPTION
**Pull request by SarahBocognano**
[FRONT | API] Add Ability to Filter Objects by Creator (#issue1512)

### Proposed changes

* Add Ability to Filter Objects by Creator (named as 'Technical Creator') and display the 'creator' columns on : 
 - Indicators
 - Reports
 - Observables
 - Vulnerabilities

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/analysis/Reports.js [hunk 1]
```diff
@@ -199,6 +199,11 @@ class Reports extends Component {
         width: '15%',
         isSortable: isRuntimeSort,
       },
+      creator: {
+        label: 'Creator',
+        width: '12%',
+        isSortable: true,
+      },
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/analysis/Reports.js:233]: Should we only filter or also display the property (same questions for the three entities) ?

**Comment by RomuDeuxfois** [comment]
So, now we can filter by creator on this entities : Report, Vulnerability and Indicator ?
It's ok for the filtering.

I can't create a new vulnerability. When I refresh I have the new entity but the screen is lock when I click on the create button.
[screen-capture (1).webm](https://user-images.githubusercontent.com/53513584/207542312-1a789e51-574e-438c-9430-59a90b273c16.webm)

--> Creation of a vulnerability is now working :)

File: opencti-platform/opencti-front/src/private/components/observations/indicators/IndicatorLine.js [hunk 2]
```diff
@@ -45,123 +43,109 @@ const styles = (theme) => ({
...
+              {node.creator.name}
+            </div>
+            <div
+              className={classes.bodyItem}
+              style={{ width: dataColumns.valid_until.width }}
+            >
+              {fd(node.valid_until)}
+            </div>
+            <div
+              className={classes.bodyItem}
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/observations/indicators/IndicatorLine.js:134]: I have a problem with the display of this property
image

File: opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableLine.js [hunk 3]
```diff
@@ -116,16 +115,18 @@ const StixCyberObservableLineComponent = (props) => {
...
+              className={classes.bodyItem}
+              style={{ width: dataColumns.creator.width }}
+            >
+              {t(node.creator.name)}
+            </div>
             <div
               className={classes.bodyItem}
               style={{ width: dataColumns.objectMarking.width }}
             >
               <ItemMarkings
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableLine.js:156]: I have a problem with the display of this property
image

File: opencti-platform/opencti-graphql/src/resolvers/indicator.js [hunk 4]
```diff
@@ -58,6 +58,10 @@ const indicatorResolvers = {
     basedOn: buildRefRelationKey(RELATION_BASED_ON),
     indicates: buildRefRelationKey(RELATION_INDICATES),
     sightedBy: buildRefRelationKey(STIX_SIGHTING_RELATIONSHIP),
+    creator: 'creator_id',
+  },
+  IndicatorsOrdering: {
+    creator: 'creator_id',
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/resolvers/indicator.js]: Do we want to order by creator? Or just filter on it ?

File: opencti-platform/opencti-front/src/private/components/analysis/Reports.js (hunk 1 above)
Comment (SarahBocognano) [reply opencti-platform/opencti-front/src/private/components/analysis/Reports.js:233]: Also display the 'creator' columns on the 4th entities

File: opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableLine.js (hunk 3 above)
Comment (SarahBocognano) [reply opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableLine.js:156]: Done !

File: opencti-platform/opencti-front/src/private/components/observations/indicators/IndicatorLine.js (hunk 2 above)
Comment (SarahBocognano) [reply opencti-platform/opencti-front/src/private/components/observations/indicators/IndicatorLine.js:134]: Done !

File: opencti-platform/opencti-graphql/src/resolvers/indicator.js (hunk 4 above)
Comment (SarahBocognano) [reply opencti-platform/opencti-graphql/src/resolvers/indicator.js]: Both :)

**Comment by RomuDeuxfois** [comment]
I have a bug when I go the report menu to show all reports.
Maybe it's just in my side...

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**